### ISSUE TITLE
NRG — Sr Software Engineer, AI Engineer (VIRTUAL, WA, US, 00000)

### ISSUE THREAD

**Issue by louisinayinde** [louisinayinde/jobs#457]
**Entreprise** : NRG  
**Poste** : Sr Software Engineer, AI Engineer (VIRTUAL, WA, US, 00000)  
**Localisation** : United States · télétravail  
**Score** : 40 — default · stack +40 (AWS, Azure, C++, GCP, Git, gRPC, Linux, NoSQL, Python, Rust, SQL)  
**Tech** : AWS, Azure, C++, GCP, Git, gRPC, Linux, NoSQL, Python, Rust, SQL  
**Lien ATS** : [himalayas.app](<https://himalayas.app/companies/nrgenergy/jobs/sr-software-engineer-ai-engineer-virtual-wa-us-00000>)  
**Publiée le** : 2026-09-19  
**Source** : himalayas  

---

**Description**

Welcome to the intersection of energy and home services. At NRG, we’re driven by our passion to create a smarter, cleaner and more connected future.

Vivint Smart Home, an NRG owned company, is a leading smart home company in the United States, dedicated to redefining the home experience with intelligent products and services. We find purpose in proactively protecting and keeping our customers connected to home, no matter where they are. Join the Smart Home team to create smarter, safer and more sustainable homes.

About This Role

As an AI Engineer on the Camera AI team, you will work alongside a group of talented and collaborative engineers to create practical AI solutions that significantly enhance home protection, provide valuable insights, and increase daily convenience. Our pioneering, fully integrated smart home ecosystem provides a unique platform to develop groundbreaking experiences in computer vision, sensor fusion, and advanced machine learning applications.

Your key responsibilities will encompass:

- Lead the design and development of high-quality, scalable AI features and products, focusing on customer benefits and technical excellence.

- Collaborate closely with cross-functional teams (embedded, apps, platform, QA, product, and UX) to successfully deploy AI solutions to our large customer base.

- Create, test, and refine advanced machine learning models, optimizing them for both edge and cloud environments.

- Develop comprehensive pipelines for data management, including curation, visualization, annotation, and model training processes.

- Continuously monitor and maintain the performance of AI features in production, ensuring ongoing customer value and discover iterative improvements.

Required Qualifications:

- Bachelor's, Master's, PhD in Data Science, Mathematics, Computer Science, Computer Engineering, Electrical Engineering or related field and 3-5 years of relevant experience Strong analytical and problem-solving skills AI/ML Experience:

- Computer vision

- Data structures, curation, annotation, and pipelines o Model training, fine-tuning, and evaluation

- Programming and Technical Proficiency: Rust, C++, Python, Shell, Embedded Linux, Git

Preferred Qualifications:

- Computer Vision: Classification, Object Detection, Segmentation, Tracking, Recognition, VLM, Pose, Vector Embeddings, Re-ID, OpenCV

- ML frameworks: TensorFlow, PyTorch

- Data Engineering:

- FiftyOne, CVAT, Vertex AI, Label Studio

- Data Visualization/Story Telling and Data Wrangling/Munging/Big Data

- MLOps/DevOps: DVC, CML, GTO, Gitlab Pipelines

- Cloud: GCP (Vertex AI/Storage/Cloud Functions), AWS, Azure

- Databases: SQL, NoSQL, Vector, Time Series, Graph

- Transformers, LLMs, Knowledge Graphs

- Multimedia pipelines: GStreamer, ffmpeg, mediapipe

- Edge computing, Embedded Systems, and IoT Devices

- Model compression techniques

- Image, Audio, Radar, Signal Processing

- Communication Protocols: gRPC, MQTT, WebRTC, Wi-Fi, Bluetooth

- Experience working with secure, scalable, high-availability, low latency, and distributed solutions

- JIRA and Confluence proficiency

NRG Energy is committed to a drug and alcohol-free workplace. To the extent permitted by law and any applicable collective bargaining agreement, employees are subject to periodic random drug testing, and post-accident and reasonable suspicion drug and alcohol testing. EOE AA M/F/Protected Veteran Status/Disability. Level, Title and/or Salary may be adjusted based on the applicant's experience or skills.

EEO is the Law Poster (The poster can be found at

Official description on file with Talent.

We support the use of AI tools to help you prepare for your interview (e.g., practicing responses, researching the role, or refining your resume). However, during interviews and assessments, we expect responses to reflect your own thinking, experience, and communication. Use of AI to generate or read answers in real time, complete assessments, or misrepresent your qualifications is not permitted and may impact your candidacy.

Originally posted on Himalayas
